core/src/main/kotlin/com/strumenta/kolasu/model/Naming.kt

```diff
@@ -49,6 +52,9 @@ class ReferenceByName<N>(val name: String, initialReferred: N? = null) : Seriali
         this.referred = initialReferred
     }
 
+    val resolved: Boolean
+        get() = referred != null
```

Comment:
Here I would say that the reference is resolved when either referred or identified are different from null. To me resolved equals false indicates that we cannot find a Node to associate to this reference, while in some cases we can find it, but we do not retrieve it straight away. The reference is still resolved, from my point of view.

I would consider adding another property called `resolvedAndRetrieved` and I would make that one checking just `referred`